// @redux-requests/core/es/actions.js
import _extends from "@babel/runtime/helpers/extends";
import { SUCCESS_SUFFIX, ERROR_SUFFIX, ABORT_SUFFIX, CLEAR_REQUESTS_CACHE, RESET_REQUESTS, ABORT_REQUESTS, SET_DOWNLOAD_PROGRESS, SET_UPLOAD_PROGRESS } from './constants';

var getActionWithSuffix = function getActionWithSuffix(suffix) {
  return function (actionType) {
    return actionType + suffix;
  };
};

export var success = getActionWithSuffix(SUCCESS_SUFFIX);
export var error = getActionWithSuffix(ERROR_SUFFIX);
export var abort = getActionWithSuffix(ABORT_SUFFIX);

var isFSA = function isFSA(action) {
  return !!action.payload;
};

export var createSuccessAction = function createSuccessAction(action, response) {
  return _extends({
    type: success(action.type)
  }, isFSA(action) ? {
    payload: response
  } : {
    response: response
  }, {
    meta: _extends({}, action.meta, {
      requestAction: action
    })
  });
};
export var createErrorAction = function createErrorAction(action, errorData) {
  return _extends({
    type: error(action.type)
  }, isFSA(action) ? {
    payload: errorData,
    error: true
  } : {
    error: errorData
  }, {
    meta: _extends({}, action.meta, {
      requestAction: action
    })
  });
};
export var createAbortAction = function createAbortAction(action) {
  return {
    type: abort(action.type),
    meta: _extends({}, action.meta, {
      requestAction: action
    })
  };
};
export var getActionPayload = function getActionPayload(action) {
  return action.payload === undefined ? action : action.payload;
}; // eslint-disable-next-line import/no-unused-modules

export var getResponseFromSuccessAction = function getResponseFromSuccessAction(action) {
  return action.payload ? action.payload : action.response;
};
export var isRequestAction = function isRequestAction(action) {
  var actionPayload = getActionPayload(action);
  return !!(actionPayload != null && actionPayload.request) && !!(Array.isArray(actionPayload.request) || actionPayload.request.url || actionPayload.request.query || actionPayload.request.promise || actionPayload.request.response || actionPayload.request.error) && !actionPayload.response && !(actionPayload instanceof Error);
};
export var isResponseAction = function isResponseAction(action) {
  var _action$meta;

  return !!((_action$meta = action.meta) != null && _action$meta.requestAction);
};
export var getRequestActionFromResponse = function getRequestActionFromResponse(action) {
  return action.meta.requestAction;
};
export var isSuccessAction = function isSuccessAction(action) {
  return isResponseAction(action) && action.type.endsWith(SUCCESS_SUFFIX);
};
export var isErrorAction = function isErrorAction(action) {
  return isResponseAction(action) && action.type.endsWith(ERROR_SUFFIX);
};
export var isAbortAction = function isAbortAction(action) {
  return isResponseAction(action) && action.type.endsWith(ABORT_SUFFIX);
};

var isRequestQuery = function isRequestQuery(request) {
  return !request.query && (!request.method || request.method.toLowerCase() === 'get') || request.query && !request.query.trim().startsWith('mutation');
};

export var isRequestActionQuery = function isRequestActionQuery(action) {
  var _action$meta2;

  var _getActionPayload = getActionPayload(action),
      request = _getActionPayload.request;

  if (((_action$meta2 = action.meta) == null ? void 0 : _action$meta2.asMutation) !== undefined) {
    return !action.meta.asMutation;
  }

  return !!(Array.isArray(request) ? request.every(isRequestQuery) : isRequestQuery(request));
};
export var clearRequestsCache = function clearRequestsCache(requests) {
  if (requests === void 0) {
    requests = null;
  }

  return {
    type: CLEAR_REQUESTS_CACHE,
    requests: requests
  };
};
export var resetRequests = function resetRequests(requests, abortPending) {
  if (requests === void 0) {
    requests = null;
  }

  if (abortPending === void 0) {
    abortPending = true;
  }

  return {
    type: RESET_REQUESTS,
    requests: requests,
    abortPending: abortPending
  };
};
export var abortRequests = function abortRequests(requests) {
  if (requests === void 0) {
    requests = null;
  }

  return {
    type: ABORT_REQUESTS,
    requests: requests
  };
};
export var setDownloadProgress = function setDownloadProgress(requestType, progress) {
  return {
    type: SET_DOWNLOAD_PROGRESS,
    requestType: requestType,
    progress: progress
  };
};
export var setUploadProgress = function setUploadProgress(requestType, progress) {
  return {
    type: SET_UPLOAD_PROGRESS,
    requestType: requestType,
    progress: progress
  };
};
