// @availity/api-axios/lib/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AvMicroserviceApi", {
  enumerable: true,
  get: function get() {
    return _ms["default"];
  }
});
Object.defineProperty(exports, "AvProxyApi", {
  enumerable: true,
  get: function get() {
    return _proxy["default"];
  }
});
Object.defineProperty(exports, "avWebQLApi", {
  enumerable: true,
  get: function get() {
    return _webQL["default"];
  }
});
Object.defineProperty(exports, "avLogMessagesApi", {
  enumerable: true,
  get: function get() {
    return _logs["default"];
  }
});
Object.defineProperty(exports, "avLogMessagesApiV2", {
  enumerable: true,
  get: function get() {
    return _dma["default"];
  }
});
Object.defineProperty(exports, "avNavigationApi", {
  enumerable: true,
  get: function get() {
    return _navigation["default"];
  }
});
Object.defineProperty(exports, "avNotificationApi", {
  enumerable: true,
  get: function get() {
    return _notification["default"];
  }
});
Object.defineProperty(exports, "avOrganizationsApi", {
  enumerable: true,
  get: function get() {
    return _organizations["default"];
  }
});
Object.defineProperty(exports, "avPermissionsApi", {
  enumerable: true,
  get: function get() {
    return _permissions["default"];
  }
});
Object.defineProperty(exports, "avProvidersApi", {
  enumerable: true,
  get: function get() {
    return _providers["default"];
  }
});
Object.defineProperty(exports, "avRegionsApi", {
  enumerable: true,
  get: function get() {
    return _regions["default"];
  }
});
Object.defineProperty(exports, "avSpacesApi", {
  enumerable: true,
  get: function get() {
    return _spaces["default"];
  }
});
Object.defineProperty(exports, "avUserApi", {
  enumerable: true,
  get: function get() {
    return _user["default"];
  }
});
Object.defineProperty(exports, "avPdfApi", {
  enumerable: true,
  get: function get() {
    return _pdf["default"];
  }
});
Object.defineProperty(exports, "avUserPermissionsApi", {
  enumerable: true,
  get: function get() {
    return _userPermissions["default"];
  }
});
Object.defineProperty(exports, "avFilesApi", {
  enumerable: true,
  get: function get() {
    return _files["default"];
  }
});
Object.defineProperty(exports, "avFilesDeliveryApi", {
  enumerable: true,
  get: function get() {
    return _filesDelivery["default"];
  }
});
Object.defineProperty(exports, "avSettingsApi", {
  enumerable: true,
  get: function get() {
    return _settings["default"];
  }
});
Object.defineProperty(exports, "avSlotMachineApi", {
  enumerable: true,
  get: function get() {
    return _slotmachine["default"];
  }
});
Object.defineProperty(exports, "avDisclaimersApi", {
  enumerable: true,
  get: function get() {
    return _disclaimers["default"];
  }
});
Object.defineProperty(exports, "avCodesApi", {
  enumerable: true,
  get: function get() {
    return _codes["default"];
  }
});
exports["default"] = void 0;

var _api = _interopRequireDefault(require("./api"));

var _ms = _interopRequireDefault(require("./ms"));

var _proxy = _interopRequireDefault(require("./proxy"));

var _webQL = _interopRequireDefault(require("./webQL"));

var _logs = _interopRequireDefault(require("./logs"));

var _dma = _interopRequireDefault(require("./dma"));

var _navigation = _interopRequireDefault(require("./navigation"));

var _notification = _interopRequireDefault(require("./notification"));

var _organizations = _interopRequireDefault(require("./organizations"));

var _permissions = _interopRequireDefault(require("./permissions"));

var _providers = _interopRequireDefault(require("./providers"));

var _regions = _interopRequireDefault(require("./regions"));

var _spaces = _interopRequireDefault(require("./spaces"));

var _user = _interopRequireDefault(require("./user"));

var _pdf = _interopRequireDefault(require("./pdf"));

var _userPermissions = _interopRequireDefault(require("./userPermissions"));

var _files = _interopRequireDefault(require("./files"));

var _filesDelivery = _interopRequireDefault(require("./filesDelivery"));

var _settings = _interopRequireDefault(require("./settings"));

var _slotmachine = _interopRequireDefault(require("./slotmachine"));

var _disclaimers = _interopRequireDefault(require("./disclaimers"));

var _codes = _interopRequireDefault(require("./codes"));

var _default = _api["default"];
exports["default"] = _default;
